### PR TITLE
Embedding module completed.

### PR DESCRIPTION
**Pull request by jisungyoon**
Eventually,  embedding module has been completed.
It will be way more faster than original implementation of Node2vec!

I added example codes and example interactive plot as well.

@ajgates42 , please take a look and approve this PR!

Thank you.

### CODE REVIEW

**Comment by jisungyoon** [comment]
@ajgates42  Can you review this PR?

File: pyscisci/embedding.py [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+    from tqdm.notebook import tqdm
+else:
+    from tqdm import tqdm
+    
+
+
+
+    
+# Node2Vec instaces    
+class Node2Vec(object):
```
Comment (ajgates42) [inline pyscisci/embedding.py:39]: Can you edit the class description to give a very brief overview of the method (ie that you first sample tons of random walks on the network, and then you embed them using Word2Vec).

File: pyscisci/embedding.py [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+        (edge_weight, (supra_edge_list[:, 0], supra_edge_list[:, 1])),
+        shape=(supra_node_num, supra_node_num),
+    )
+    return Aspra, src_trg_pairs
+
+
+# sim_axis_code
+# For the detail, please read Sem_aixs paper, https://arxiv.org/abs/1806.05521. <br>
+
+def sem_axis(emb, positive_entities, negative_entities):
```
Comment (ajgates42) [inline pyscisci/embedding.py:430]: Please briefly document the sem_axis function.

File: pyscisci/embedding.py [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+"""
+.. module:: embedding
+    :synopsis: The main Embedding Class
+
+.. moduleauthor:: Jisung Yoon <jisung.yoon92@gmail.com>
+ """
+import sys
+import numpy as np
+from scipy import sparse
+import networkx as nx
```
Comment (ajgates42) [inline pyscisci/embedding.py:11]: move networkx to optional dependency, this is the only function to use it (and you really only convert out of nx graphs).

File: pyscisci/embedding.py [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+except ImportError:
+    raise ImportError('Please install numba to take full advantage of fast embedding tools. \n pip install numba')
+    
+try:    
+    from gensim.models import Word2Vec
+except ImportError:
+    raise ImportError('Please install gensim to take full advantage of fast embedding tools. \n pip install gensim')
+
+# determine if we are loading from a jupyter notebook (to make pretty progress bars)
+if 'ipykernel' in sys.modules:
```
Comment (ajgates42) [inline pyscisci/embedding.py:29]: you can drop this dependency here; there are no calls to tqdm

**Comment by ajgates42** [comment]
Overall it looks great, just a few minor suggestions on dependencies and documentation.

Also, can you add some of your references to the bib file in the doc?

**Comment by jisungyoon** [comment]
I just covered all comments!  Sorry for late, I was at IC2S2.

Btw, the last commit is the result of code formatted (black and isort), it automatically styles the code.
If you don't mind, I can run the code formatter for all codes in this repository at another PR!

**Comment by jisungyoon** [comment]
@ajgates42  can you review this pr?
Thanks!

**Pull request merged**